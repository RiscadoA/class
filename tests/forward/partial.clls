proc main() {
  cut {
    send x(a. println("wait a"); wait a; ());
    send x(b. println("wait b"); wait b; ());
    wait x; println("wait x"); ()
    |x: recv close; recv close; close|
    recv x(a);
    par {
      println("close a"); close a
      ||
      println("fwd x y"); fwd x y
    }
    |y: recv close; close|
    cut {
      recv y(b);
      par {
        println("close b"); close b
        ||
        println("fwd y z"); fwd y z
      }
      |z: close|
      println("close z"); close z
    }
  }
};;
