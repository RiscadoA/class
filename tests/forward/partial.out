close a
wait a
fwd x y
close z
close b
wait b
fwd y z
wait x
